### PR TITLE
Issue 2060 StringAssert: add variants with "Ignoring"

### LINKED ISSUE #2060: StringAssert: add variants with "Ignoring"
#### Summary

Currently the `AbstractCharSequenceAssert` has a method `containsIgnoringCase(...)`. It would appreciable to have the same variants as the `isEqualToIgnoring...`. So:

- `containsIgnoringWhitespaces(CharSequence sequence)`
- `containsIgnoringNewLines(CharSequence sequence)`

#### Example

```java
assertThat("gandalf is a \r\nwizard").containsIgnoringNewLines("gandalf is a wizard") // SUCCESS
assertThat("gandalf is a wizard").containsIgnoringNewLines("gandalf is a \nwizard") // SUCCESS
```

### PR DESCRIPTION
**Pull request by spring1627**
#### Check List:
* Fixes #2060 (partially)
* Unit tests : YES
* Javadoc with a code example (on API only) : YES
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md)

Following the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md) will make it easier for us to review and accept your PR.

### CODE REVIEW

**Comment by joel-costigliola** [comment]
Thanks, we will review your PR shortly!

File: src/main/java/org/assertj/core/internal/Strings.java [hunk 1]
```diff
@@ -527,6 +529,51 @@ public void assertContainsIgnoringCase(AssertionInfo info, CharSequence actual,
...
+   * @throws NullPointerException if the given sequence is {@code null}.
+   * @throws IllegalArgumentException if the given values is empty.
+   * @throws AssertionError if the given {@code CharSequence} is {@code null}.
+   * @throws AssertionError if actual {@code CharSequence} doesn't have sequence
+   */
+  public void assertContainsIgnoringNewLines(final AssertionInfo info, final CharSequence actual, final CharSequence... values) {
+    doCommonCheckForCharSequence(info, actual, values);
+    final String actualNoNewLines = removeNewLines(actual);
+    final Stream<CharSequence> notFoundStream = stream(values);
+    final Stream<CharSequence> notFoundStream2 = mapContainsIgnoringNewLines(notFoundStream);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Strings.java:547]: inlining both `notFoundStream` and `mapContainsIgnoringNewLines` would make the code clearer I think, here it's confusing to have `notFoundStream` and `notFoundStream2`.
might look like:
```java
Set<CharSequence> notFound = stream(values).map(Strings::removeNewLines)
                                           .filter(value -> !stringContains(actualNoNewLines, value))
                                           .collect(toCollection(LinkedHashSet::new));
```

File: src/main/java/org/assertj/core/internal/Strings.java [hunk 2]
```diff
@@ -527,6 +529,51 @@ public void assertContainsIgnoringCase(AssertionInfo info, CharSequence actual,
...
+  private Stream<CharSequence> filterContainsIgnoringNewLines(final String actualNoNewLines,
+                                                              final Stream<CharSequence> notFoundStream) {
+    return notFoundStream.filter(value -> !stringContains(actualNoNewLines, value));
+  }
+
+  private Set<CharSequence> collectContainsIgnoringNewLines(final Stream<CharSequence> notFoundStream) {
+    return notFoundStream.collect(toCollection(LinkedHashSet::new));
+  }
+
+  private boolean isEmptyContainsIgnoringNewLines(final Set<CharSequence> notFound) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Strings.java:573]: inline this as the inline code is as clear as this method name

File: src/test/java/org/assertj/core/api/charsequence/CharSequenceAssert_containsIgnoringNewLines_CharSequence_Test.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * CharSequenceAssert#containsIgnoringNewLines(CharSequence...)}
+ * (CharSequence...)}</code>.
+ */
+class CharSequenceAssert_containsIgnoringNewLines_CharSequence_Test extends CharSequenceAssertBaseTest {
+  @Override
+  protected CharSequenceAssert invoke_api_method() {
+    CharSequenceAssert result;
+    try {
+      result = assertions.containsIgnoringNewLines("Al", "Bob");
+    } catch (CloneNotSupportedException e) {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/charsequence/CharSequenceAssert_containsIgnoringNewLines_CharSequence_Test.java:32]: no need to deal with CloneNotSupportedException

File: src/test/java/org/assertj/core/internal/strings/Strings_assertContainsIgnoringNewLines_Test.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * Tests for <code>{@link Strings#assertContainsIgnoringNewLines(AssertionInfo, CharSequence, CharSequence...)} </code>.
+ */
+public class Strings_assertContainsIgnoringNewLines_Test extends StringsBaseTest {
+  private static final WritableAssertionInfo INFO = someInfo();
+
+  @ParameterizedTest
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/strings/Strings_assertContainsIgnoringNewLines_Test.java:28]: nice!

**Review by joel-costigliola — CHANGES_REQUESTED** [review]
Thanks @spring1627 , first round of code review done.

File: src/main/java/org/assertj/core/internal/Maps.java [hunk 5]
```diff
@@ -504,6 +498,8 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
   }
 
   public <K, V> void assertContainsExactly(AssertionInfo info, Map<K, V> actual, Entry<? extends K, ? extends V>[] entries) {
+    System.out.println(actual.toString());
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Maps.java:501]: remove all System.out, better debugging with an IDE ;-)

File: src/test/java/org/example/custom/map_test.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import static org.assertj.core.util.Arrays.array;
+import static org.assertj.core.data.MapEntry.entry;
+
+public class map_test extends MapsBaseTest {
+
+  @VisibleForTesting
+  Maps maps = Maps.instance();
+
+
+  // ============================= MY TEST HERE ===================================================
```
Comment (joel-costigliola) [inline src/test/java/org/example/custom/map_test.java:24]: remove

File: src/test/java/org/example/custom/map_test.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.assertj.core.internal.Maps;
+import org.apache.commons.collections4.map.CaseInsensitiveMap;
+import static org.assertj.core.util.Arrays.array;
+import static org.assertj.core.data.MapEntry.entry;
+
+public class map_test extends MapsBaseTest {
```
Comment (joel-costigliola) [inline src/test/java/org/example/custom/map_test.java:18]: use proper naming convention (look at the other test classes if you are not sure)

Comment (joel-costigliola) [reply src/test/java/org/example/custom/map_test.java:18]: and please follow the contributing guide https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md thanks!

**Review by joel-costigliola — CHANGES_REQUESTED** [review]
please address the first comments we have before we can review this PR fully, thanks

File: src/test/java/org/assertj/core/internal/files/Files_assertHasDigest_AlgorithmString_Test.java [hunk 8]
```diff
@@ -38,17 +41,18 @@
...
 import org.assertj.core.internal.FilesBaseTest;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 /**
  * Tests for <code>{@link Files#assertHasDigest(AssertionInfo, File, MessageDigest, String)}</code>
  *
  * @author Valeriy Vyrva
  */
+@DisplayName("Files.assertHasDigest_AlgorithmString:")
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/files/Files_assertHasDigest_AlgorithmString_Test.java:52]: no need to add any `DisplayName` as they are automatically computed

File: src/test/java/org/assertj/core/internal/files/File_assertHasSize_Test.java [hunk 9]
```diff
@@ -57,8 +57,9 @@ void should_throw_error_if_actual_file_does_not_have_the_expected_size() {
     // GIVEN
     AssertionInfo info = someInfo();
     // WHEN
-    expectAssertionError(() -> files.assertHasSizeInBytes(info, actual, 36L));
+    AssertionError error = expectAssertionError(() -> files.assertHasSizeInBytes(info, actual, 36L));
     // THEN
+    assertThat(error).isInstanceOf(AssertionError.class);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/files/File_assertHasSize_Test.java:62]: `expectAssertionError` already makes sure the error is an `AssertionError`, please revert all the changes in this file.
The only thing to do is remove @DisplayName("Files.assertHasSizeInBytes:") as it is computed automatically

**Comment by joel-costigliola** [comment]
@ahmadsadeed could you address the comments in this PR before adding more code? thanks!

**Comment by scordio** [comment]
Something is strange with this PR, files unrelated to the original issue are having changes.

@spring1627 could you please have a look at the commits of this branch and push a clean history?

**Comment by ahmadsadeed** [comment]
> Something is strange with this PR, files unrelated to the original issue are having changes.
> 
> @spring1627 could you please have a look at the commits of this branch and push a clean history?

Is it the files starting with Files_ ? I was working on issue #2337 and pushed my commit to our group [repo](https://github.com/sandeep-yerramsetty/assertj-core)  . My commit was not supposed to be included in this PR because I made a separate [PR](https://github.com/assertj/assertj-core/pull/2422) from my Github account. My commit was merged with the main branch where @spring1627 pushed their code for issue 2060. I suggest @spring1627 make a new commit with changes only for issue 2060 from their github account. I will push my changes to issue 2337 on my forked [repo](https://github.com/assertj/assertj-core/pull/2422). This way each PR will be specific to one issue.

**Comment by joel-costigliola** [comment]
closing this PR as it was mixed up with another one

**Pull request closed without merging**